core/commonKotlin/src/Instant.kt

```diff
@@ -106,13 +119,15 @@ public actual class Instant internal constructor(public actual val epochSeconds:
         public actual fun now(): Instant = currentTime()
 
         // org.threeten.bp.Instant#ofEpochMilli
-        public actual fun fromEpochMilliseconds(epochMilliseconds: Long): Instant =
-            if (epochMilliseconds < MIN_SECOND * MILLIS_PER_ONE) MIN
-            else if (epochMilliseconds > MAX_SECOND * MILLIS_PER_ONE) MAX
-            else Instant(
-                epochMilliseconds.floorDiv(MILLIS_PER_ONE.toLong()),
-                (epochMilliseconds.mod(MILLIS_PER_ONE.toLong()) * NANOS_PER_MILLI).toInt()
-            )
+        public actual fun fromEpochMilliseconds(epochMilliseconds: Long): Instant {
+            val epochSeconds = epochMilliseconds.floorDiv(MILLIS_PER_ONE.toLong())
+            val nanosecondsOfSecond = (epochMilliseconds.mod(MILLIS_PER_ONE.toLong()) * NANOS_PER_MILLI).toInt()
+            return when {
+                epochSeconds < MIN_SECOND -> MIN
+                epochSeconds > MAX_SECOND -> MAX
```

Comment:
Is it still possible? If the Instant range doesn't potentially fit in the Long range of milliseconds, doesn't it mean that the Long range of milliseconds always fit in the Instant range?